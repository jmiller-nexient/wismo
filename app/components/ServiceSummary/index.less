@import '../../styles/themes/index.less';

@service-summary-prefix-cls: ~'@{dte-prefix}-service-summary';

.@{service-summary-prefix-cls} {
  border: 1px solid @dte-gray-400;
  column-count: 2;
  margin: 5px;
  -moz-column-count: 2;
  padding: 10px;
  -webkit-column-count: 2;
  @media screen and (max-width: 400px) {
    column-count: 1;
    -moz-column-count: 1;
    -webkit-column-count: 1;
  }
}
