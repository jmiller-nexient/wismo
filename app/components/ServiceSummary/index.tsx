import * as React from 'react';

import AccountNumberFormField from 'components/AccountNumberField';
import AddressFormField from 'components/AddressField';
import ProductsFormField from 'components/ProductsListField';

import IServiceRequest from 'types/serviceRequest';

import './index.less';

interface IServiceSummaryProps {
  serviceRequest: IServiceRequest;
}

const ServiceSummary: React.FC<IServiceSummaryProps> = (props: IServiceSummaryProps): React.ReactElement => {
  const { accountNumber, address, products } = props.serviceRequest;

  return (
    <div className={'dte-wismo-service-summary'}>
      <AddressFormField address={address}/>
      <AccountNumberFormField accountNumber={accountNumber}/>
      <ProductsFormField products={products}/>
    </div>
  );
};

export default ServiceSummary;
