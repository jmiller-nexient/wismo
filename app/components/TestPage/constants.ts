export const fakeProductList = [
  'Residential Gas Heating',
  'Residential Electric Service',
  'Interuptable Air Conditioning'
];
