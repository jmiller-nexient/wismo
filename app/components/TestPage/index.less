@import '../../styles/themes/index.less';

@testpage-prefix-cls: ~'@{dte-prefix}-testpage';

.@{testpage-prefix-cls} {
  .customer-name {
    margin-bottom: 0;
  }
  .edit-button {
    background-color: @dte-white;
    border: none;
    color: @dte-blue-light;
    font-size: @dte-wismo-font-size-0;
    margin-left: 10px;
    outline: none;
    padding: 0;
    text-decoration: underline;
    &:hover:enabled {
      background-color: @dte-white;
    }
  }
  .indented {
    margin-left: 20px;
  }
  .phone-number {
    color: @dte-gray-600;
    margin-top: 8px;
  }
}
