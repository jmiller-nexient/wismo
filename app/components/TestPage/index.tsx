import * as React from 'react';

import AccountNumber from 'components/AccountNumber';
import Button from 'components/Button';
import Currency from 'components/Currency';
import CustomerName from 'components/CustomerName';
import DateStamp from 'components/DateStamp';
import Header from 'components/Header';
import Hyperlink from 'components/Hyperlink';
import OrderNumber from 'components/OrderNumber';
import PhoneNumber from 'components/PhoneNumber';
import ProductList from 'components/ProductList';
import SelectLanguage from 'components/SelectLanguage';

import { fakeProductList } from './constants';

import './index.less';

const TestPage: React.FC = (): React.ReactElement => (
  <div className="dte-wismo-testpage">
    <Header />
    <h1>This is the test\demo page for custom components</h1>
    <div>
      <SelectLanguage />
    </div>
    <div>
      <h4>Account Number component:</h4>
      <div className="indented">
        <AccountNumber value="927530986753" />
      </div>
    </div>
    <div>
      <h4>Order Number component:</h4>
      <div>
        <OrderNumber value="123456789012" />
      </div>
    </div>
    <div>
      <h4>Button component:</h4>
      <div className="indented">
        <Button type="primary">Primary</Button>
        &nbsp;
        <Button disabled type="primary">
          Primary (Disabled)
        </Button>
        &nbsp;
        <Button type="secondary">Secondary</Button>
        &nbsp;
        <Button disabled type="secondary">
          Secondary (Disabled)
        </Button>
      </div>
    </div>
    <div>
      <h4>Customer Name component:</h4>
      <div className="indented">
        <CustomerName firstName="Adam" lastName="Naglich" />
      </div>
    </div>
    <div>
      <h4>Hyperlink component:</h4>
      <div className="indented">
        Internal URL: <Hyperlink href="/">Home</Hyperlink>
        <br />
        External URL: <Hyperlink href="http://www.dteenergy.com">DTE Energy</Hyperlink>
      </div>
    </div>
    <div>
      <h4>Phone Number component:</h4>
      <div className="indented">
        7-Digits: <PhoneNumber value="1234567" />
        <br />
        10-Digits: <PhoneNumber value="1234567890" />
      </div>
    </div>
    <div>
      <h4>Date Stamp component:</h4>
      <div className="indented">
        Long/Default: <DateStamp date="2020/04/13" />
        <br />
        Short: <DateStamp date="2020/04/13" short />
        <br />
        ISO Date Long: <DateStamp date="2020-03-19T13:15:30Z" />
        <br />
        ISO Date Short: <DateStamp date="2020-03-19T13:15:30Z" short />
        <br />
        Invalid Date: <DateStamp date="Tuesday" />
      </div>
    </div>
    <div>
      <h4>Product List component:</h4>
      <div className="indented">
        <ProductList products={fakeProductList} />
      </div>
    </div>
    <div>
      <h4>Currency component:</h4>
      <div className="indented">
        <Currency value={1000} />
      </div>
    </div>
    <div>
      <h4>Currency component:</h4>
      <div className="indented">
        <Currency value={-1.23} />
      </div>
    </div>
    <div>
      <h4>Currency component:</h4>
      <div className="indented">
        <Currency value=".59" />
      </div>
    </div>
    <div>
      <h4>Currency component:</h4>
      <div className="indented">
        <Currency value="1" />
      </div>
    </div>
    <div>
      <h4>Customer Info:</h4>
      <h4 className="customer-name indented">
        <CustomerName firstName="Adam" lastName="Naglich" />
      </h4>
      <p className="indented phone-number">
        <PhoneNumber value="1234567890" />
        <button className="edit-button" type="button">
          Edit number
        </button>
      </p>
    </div>
  </div>
);

export default TestPage;
