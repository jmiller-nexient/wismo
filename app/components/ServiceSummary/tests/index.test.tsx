import React from 'react';
import renderer from 'react-test-renderer';

import ServiceSummary from 'components/ServiceSummary';

import { fakeServiceRequest } from 'components/TestPage/constants';
import { IntlProvider } from 'react-intl';

const serviceRequestNoProducts = {
  ...fakeServiceRequest,
  products: [],
};

describe('<ServiceSummary />', () => {
  it('should render correctly when the page loads', () => {
    const tree = renderer.create(
      <IntlProvider locale={'en'}>
        <ServiceSummary serviceRequest={fakeServiceRequest}/>
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when product list is empty', () => {
    const tree = renderer.create(
      <IntlProvider locale={'en'}>
        <ServiceSummary serviceRequest={serviceRequestNoProducts}/>
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
