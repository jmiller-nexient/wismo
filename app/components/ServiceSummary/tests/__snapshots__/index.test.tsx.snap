// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceSummary /> should render correctly when product list is empty 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Address
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
          <br />
          Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Account Number
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      >
        <span>
          1111 222 3333 4
        </span>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Products
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      />
    </div>
  </div>
</div>
`;

exports[`<ServiceSummary /> should render correctly when the page loads 1`] = `
<div
  className="dte-wismo-service-summary"
>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Address
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      >
        <address
          className="dte-wismo-address"
        >
          123 Main
          <br />
          Bloomfield, MI 48301
        </address>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Account Number
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      >
        <span>
          1111 222 3333 4
        </span>
      </span>
    </div>
  </div>
  <div
    className="dte-wismo-base-field-section"
  >
    <label
      className="dte-wismo-base-field-label"
    >
      Products
    </label>
    <div>
      <span
        className="dte-wismo-base-field-children"
      >
        Residential Gas Heating
        , 
        Residential Electric Service
        , 
        Interuptable Air Conditioning
      </span>
    </div>
  </div>
</div>
`;
